common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/OAuth2Strategy.java

```diff
@@ -230,6 +245,101 @@ protected final void setAuthorizationEndpoint(final String authorizationEndpoint
         mAuthorizationEndpoint = authorizationEndpoint;
     }
 
+    public AuthorizationResult getDeviceCode(MicrosoftStsAuthorizationRequest authorizationRequest, @Nullable String authorityUrl) throws IOException {
+        final String methodName = ":getDeviceCode";
+
+        // Set up connection
+        String urlBody;
+
+        if (authorityUrl == null) {
+            // Fetch the tenant from the authority
+            String tenant = authorizationRequest.getAuthority().getPath().split("/")[1];
+
+            // Set up Device Code Flow authorization url
+            urlBody = "https://login.microsoftonline.com/" + tenant + "/oauth2/v2.0/devicecode";
+        }
+        else {
+            urlBody = authorityUrl;
+        }
+
+        URL url = new URL(urlBody);
+
+        final HttpURLConnection con = (HttpsURLConnection) url.openConnection();
```

Comment:
Networking code shouldn't be at this level of abstraction; instead delegate to existing implementation in `HttpRequest.java`

Reply:
Will convert to use `HttpRequest`

Reply:
> Will convert to use `HttpRequest`

Better, make an HttpClient from the same package and use it.  We can replace your member field with an injection point in the future.

Reply:
@AdamBJohnsonx I've already pushed a version using HttpRequest. I can go back and update it to use HttpClient once a few of my other tasks are complete.